serializers/src/glTF/2.0/babylon.glTFMaterial.ts

```diff
@@ -392,12 +406,237 @@ module BABYLON.GLTF2 {
+                        maxBaseColor.g = Math.max(maxBaseColor.g, metallicRoughness.baseColor.g);
+                        maxBaseColor.b = Math.max(maxBaseColor.b, metallicRoughness.baseColor.b);
+                        maxMetallic = Math.max(maxMetallic, metallicRoughness.metallic);
+                        maxRoughness = Math.max(maxRoughness, metallicRoughness.roughness);
+
+                        baseColorBuffer[offset] = metallicRoughness.baseColor.r * 255;
+                        baseColorBuffer[offset + 1] = metallicRoughness.baseColor.g * 255;
+                        baseColorBuffer[offset + 2] = metallicRoughness.baseColor.b * 255;
+                        const alpha = resizedTextures.texture1.hasAlpha ? diffuseBuffer[offset + 3] : 255;
+                        baseColorBuffer[offset + 3] = alpha;
+
+                        metallicRoughnessBuffer[offset] = 0;
+                        metallicRoughnessBuffer[offset + 1] = metallicRoughness.roughness * 255;
+                        metallicRoughnessBuffer[offset + 2] = metallicRoughness.metallic * 255;
+                        metallicRoughnessBuffer[offset + 3] = 255;
+                    }
+                }
+
+                // Retrieves the metallic roughness factors from the maximum texture values.
+                const baseColorDiff = Color3.White().subtract(maxBaseColor);
+                const metallicDiff = maxMetallic > this.epsilon ? 1 - maxMetallic : 0;
+                const roughnessDiff = maxRoughness > this.epsilon ? 1 - maxRoughness : 0;
+
+                const metallicRoughnessFactors: _IPBRMetallicRoughness = {
+                    baseColor: maxBaseColor,
+                    metallic: 1 - metallicDiff,
+                    roughness: 1 - roughnessDiff
+                };
+
+                for (let h = 0; h < height; ++h) {
+                    for (let w = 0; w < width; ++w) {
+
+                        const destinationOffset = (width * h + w) * strideSize;
+
+                        if (!baseColorDiff.equalsFloats(0, 0, 0)) {
+
+                            baseColorBuffer[destinationOffset] += baseColorDiff.r * 255;
+                            baseColorBuffer[destinationOffset + 1] += baseColorDiff.g * 255;
+                            baseColorBuffer[destinationOffset + 2] += baseColorDiff.b * 255;
+                        }
+
+                        metallicRoughnessBuffer[destinationOffset + 1] += roughnessDiff * 255;
+                        metallicRoughnessBuffer[destinationOffset + 2] += metallicDiff * 255;
+                    }
+                }
+
+                const metallicRoughnessBase64 = this.CreateBase64FromCanvas(metallicRoughnessBuffer, width, height, mimeType);
+                const baseColorBase64 = this.CreateBase64FromCanvas(baseColorBuffer, width, height, mimeType);
+
+                metallicRoughnessFactors.metallicRoughnessTextureBase64 = metallicRoughnessBase64;

```

Comment:
We discusses some things offline. Need to fix max factors math and investigate alpha issues.